riscv/src/compiler.rs

```diff
@@ -202,6 +206,25 @@ fn replace_dynamic_label_reference(
     ))
 }
 
+fn replace_coprocessor_stubs(statements: &mut Vec<Statement>) {
+    let mut to_delete = BTreeSet::default();
+    for (i, statement) in statements.iter().enumerate() {
+        if let Statement::Label(label) = statement {
+            let stub_names: Vec<&str> = COPROCESSOR_SUBSTITUTIONS
+                .iter()
+                .map(|&(ref name, _)| *name)
+                .collect();
+
+            if stub_names.contains(&label.as_str()) {
+                to_delete.insert(i);        // for the label
+                to_delete.insert(i + 1);    // for the `ret` instruction
+            }
+        }
+    }
+    let mut i = 0;
+    statements.retain(|_| (!to_delete.contains(&i), i += 1).0);
```

Comment:
This is a bit obscure to me. How about something like [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f5a1a35a9d84adb7e81a59b0171636f7) for this whole function?

Reply:
I adapted it from something I found in the same file, but sure that would work too;